dispatch/apps/api/views.py

```diff
@@ -407,6 +407,42 @@ def articles(self, request, pk=None):
         view = ArticleViewSet.as_view({'get': 'topic'})
         return view(request, pk=pk)
 
+class FileViewSet(viewsets.ModelViewSet):
+    """
+    Viewset for File model views.
+    """
+    model = File
+    serializer_class = FileSerializer
+
+    def get_queryset(self):
+        queryset = File.objects.all()
+        q = self.request.query_params.get('q', None)
+        if q is not None:
+            # If a search term (q) is present, filter queryset by term against `name`
+            queryset = queryset.filter(name__icontains=q)
+        return queryset
+
+    def bulk_delete(self, request):
+        deleted = []
+        ids = self.request.data.get('ids', None)
```

Comment:
`None` is the default value returned for `dict.get(nonexistent-key)`. It's generally more idiomatic to omit it:
```py
ids = self.request.data.get('ids')
```

This whole statement can be simplified into something like
```py
ids = self.request.data.get('ids', '').split(',')
File.objects.filter(id__in=ids).delete()

data = {
    'deleted': map(int, ids)
}
```

Does the frontend actually use the list of deleted IDs? What would be a case where only some deletions would fail?